sigle/src/pages-lib/_error.tsx

```diff
@@ -1,38 +1,136 @@
-import React from 'react';
+import React /*{ useEffect }*/ from 'react';
+import { useEffect } from 'react';
 import { NextPageContext } from 'next';
 import NextErrorComponent from 'next/error';
-import styled from 'styled-components';
-import tw from 'twin.macro';
+import { styled } from '../stitches.config';
 import Link from 'next/link';
 import * as Sentry from '@sentry/nextjs';
-import { Button, Container } from '../components';
-import { sigleConfig } from '../config';
+import { Button, Text } from '../ui';
+import { Container } from '../ui';
+import Image from 'next/image';
+import { useAuth } from '../modules/auth/AuthContext';
 
-const NotFoundContainer = styled(Container)`
-  ${tw`mt-8 mb-8 flex flex-col items-center text-center md:flex-row md:text-left`};
-`;
+const NotFoundContainer = styled(Container, {
+  '@xs': {
+    flexDirection: 'column',
+    textAlign: 'center',
+  },
 
-const NotFoundTextContainer = styled(Container)`
-  ${tw`flex flex-col items-center md:items-start md:pl-8`};
-`;
+  '@lg': {
+    flexDirection: 'row-reverse',
+    textAlign: 'left',
+  },
 
-const NotFoundTitle = styled.div`
-  ${tw`mb-2 text-5xl`};
-`;
+  px: 0,
+  width: '100%',
+  display: 'flex',
+  alignItems: 'center',
+  mx: 'auto',
+  mt: '$8',
+  mb: '$3',
+});
 
-const NotFoundSubTitle = styled.div`
-  ${tw`mb-8 text-3xl`};
-`;
+const NotFoundTextContainer = styled(Container, {
+  '@xs': {
+    alignItems: 'center',
+  },
 
-const NotFoundIllu = styled.img`
-  ${tw`mt-4 mb-4`};
-  width: 300px;
-  max-width: 100%;
+  '@md': {
+    width: '501px',
+  },
 
-  @media (min-width: ${sigleConfig.breakpoints.md}px) {
-    width: 520px;
-  }
-`;
+  '@lg': {
+    alignItems: 'start',
+  },
+
+  px: 0,
+  pr: '$5',
+  display: 'flex',
+  flexDirection: 'column',
+});
+
+const NotFoundTitle = styled('h1', {
+  mb: '$3',
+  lineHeight: 1,
+
+  fontSize: '$9',
+  fontWeight: 700,
+});
+
+const NotFoundSubTitle = styled('h3', {
+  mb: '$6',
+  fontSize: '$5',
+  fontWeight: 700,
+  lineHeight: 1,
+});
+
+const NotFoundText = styled(Text, {
+  mb: '$6',
+  maxWidth: '400px',
+});
+
+const NotFoundWrapper = styled('div', {
+  '@xs': {
+    width: '300px',
+    height: '282px',
+  },
+
+  '@lg': {
+    width: '520px',
+    height: '488px',
+  },
+
+  '@2xl': {
+    width: '737px',
+    height: '693px',
+  },
+
+  position: 'relative',
+  mt: '$4',
+  mb: '$4',
+  maxWidth: '100%',
+});
+
+const ErrorWrapper = styled('div', {
+  '@xs': {
+    width: '300px',
+    height: '232px',
+  },
+
+  '@lg': {
+    width: '520px',
+    height: '403px',
+  },
+
+  '@2xl': {
+    width: '765px',
+    height: '593px',
+  },
+
+  position: 'relative',
+  mt: '$4',
+  mb: '$4',
+  maxWidth: '100%',
+});
+
+const NotFoundIllu = ({
+  statusCode,
+  alt,
+}: {
+  statusCode: number;
+  alt: string;
+}) => (
+  <Image
+    src={`/static/img/${statusCode === 404 ? '404' : '5XX'}.png`}
```

Comment:
Are dynamic src values on the Image component allowed in the Next.js? I thought this was forbidden so the image paths can be extracted at build time. Not 100% sure about this tbh and I am not able to find a clear answer in the documentation.